### PR TITLE
feat: Relay Integration tests

### PR DESCRIPTION
**Pull request by Elyniss**

### CODE REVIEW

File: .github/workflows/ci_relay.yml [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
+
+on:
+  push:
+    branches:
+      - develop
+      - master
+  pull_request:
+    branches:
+      - develop
+      - master
```
Comment (TalhaAli00) [inline .github/workflows/ci_relay.yml:11]: Do we want to run this on develop?

**Review by TalhaAli00 — APPROVED** [review]

File: .github/workflows/ci_relay.yml (hunk 1 above)
Comment (jakobuid) [reply .github/workflows/ci_relay.yml:11]: For me it depends on the time. How long does it take?

File: foundation/src/test/kotlin/com/walletconnect/foundation/RelayTest.kt [hunk 2]
```diff
@@ -5,29 +5,133 @@ import com.walletconnect.foundation.di.commonModule
...
-                commonModule(),
-                cryptoModule()
-            )
+    fun `Single test`() {
+        // todo: Remove later on. Consolidate to one function below
+        `One client sends unencrypted message, second one receives it`()
+    }
+
+    @ExperimentalTime
+    // todo: Add @Test annotation later on.
```
Comment (jakobuid) [inline foundation/src/test/kotlin/com/walletconnect/foundation/RelayTest.kt:45]: Why this annotation is a problem?

**Review by jakobuid — APPROVED** [review]

File: .github/workflows/ci_relay.yml (hunk 1 above)
Comment (Elyniss) [reply .github/workflows/ci_relay.yml:11]: @jakobuid Most recent took 1,5 min: https://github.com/WalletConnect/WalletConnectKotlinV2/actions/runs/2962097261

File: foundation/src/test/kotlin/com/walletconnect/foundation/RelayTest.kt (hunk 2 above)
Comment (Elyniss) [reply foundation/src/test/kotlin/com/walletconnect/foundation/RelayTest.kt:45]: I moved this test to a separate function so I can showcase current issues on our rs-relay, which are not found in ts-relay. Later on we should add `@Test` annotation here and remove tests: `N consequent tests` and `Single test`

File: .github/workflows/ci_relay.yml (hunk 1 above)
Comment (Elyniss) [reply .github/workflows/ci_relay.yml:11]: And another sample which took 2,5 min: https://github.com/WalletConnect/WalletConnectKotlinV2/actions/runs/2962142651

Comment (Elyniss) [reply .github/workflows/ci_relay.yml:11]: In conclusion between 1,5 min to 3 min. Also, I added caching to our unit tests, which shorten the execution time from ~6-7 min to 2~3 min.

**Pull request merged**